### DISCUSSION TITLE
Setting other field values

### DISCUSSION THREAD

**Discussion by RayZorb** [bellrichm/WeeWX-MQTTSubscribe#180] [Q&A]
Hi,
I have an AS3935 lightning detector sending this information:

"epochtime": 1680431033, "timestamp": "2023-04-02 20:23:53", "event_type": "sim_strike", "distance": 19, "energy": 1002729}

I need to seperate the event_type to be updating different fields in whew depending on the event_type value.

ie, if event_type = "noise" then add a 1 to lightning_noise_count
if event_type = "disturber" then add a 1 to lightning_disturber_count
if event_type = "strike" update the corresponding values for distance/energy and add 1 to strike count.

Here is my weewx.conf:

[MQTTSubscribeService]

    enable = true
    host = 192.168.0.253
    port = 1883
    keepalive = 60
    username = None
    password = None

    binding = loop

    use_server_datetime = True
    #archive_interval = 10
    ignore_start_time = true
    ignore_end_time = true
    log = true
    # This section is for the MQTTSubscribe service.

    [[message_callback]]
        type = json

    [[topics]]
        ignore = True
        unit_system = METRIC

        [[[as3935/strike_update]]]
            [[[[epochtime]]]]
                ignore = True

            [[[[timestamp]]]]
                ignore = True

            [[[[event_type]]]]
                ignore = True
                # Need to add some logic here.....

            [[[[distance]]]]
                ignore = False
                name = lightning_distance

            [[[[energy]]]]
                ignore = False
                name = lightning_energy

**Comment by bellrichm** [comment]
Unfortunately what you are asking to do is not possible. But, there might be other ways to accomplish it. 
In the WeeWX google group, I have seen people use the StdCalibrate section to do some calculations.

When you say 'add a 1 to lightning_noise_count', what exactly do you mean by that? In general I would think you just want the lightning_noise_count to be the count for your archive period and then use WeeWX aggregation functions to display a count for day, week, etc. 

There is some logic in MQTTSubscribe to rename fields based on other fields. See, https://github.com/bellrichm/WeeWX-MQTTSubscribe/wiki/Configuring-additional-options#msg_id_field 
But, I am not sure you really need to add this complexity... If the event is of type noise, what is the incoming value of a field like lightning_noise_count? Is it the value from the previous noise event, which is why you need to ignore it?

Did you bind to the loop or archive? What is your archive interval?

I don't have any lightning detection hardware, so this is new to me. If you want to post more detailed MQTT data we might be able to come up with a solution.
